File: Kotlin/src/driver.kt
fun print_meeting(meeting: Meeting): Unit {
  println("\nMeeting Scheduled\n\tMeeting name: ${meeting.event_name}\n\tStart Time: ${meeting.start_time}\n\tEnd Time: ${meeting.end_time}\n\tDuration: ${meeting.duration()}\n\tDate: ${meeting.date}\n\tAttendance: ${meeting.participant_list.size} People\n\tAttendees: ${meeting.participant_list.joinToString(", ")}\n\n")
}

fun main(args: Array<String>) {
  val meeting1 = schedule meeting {
    meeting name "Making Code Simple Again"
    starts at 10.30
    and ends 12.30
    date = 12 April 1212
    participants include "Venkat." + "Venkat Again." + "Also Venkat."
  }

  val meeting2 = schedule meeting {
    meeting name "Meeting of Odd Named Individuals"
    starts at 14.30
    and ends 18.00
    date = 23 October 2018
    participants include "Sam." + "Man." + "Him." + "Jim." + "Slim."
  }

  print_meeting(meeting1)
  print_meeting(meeting2)
}

File: Ruby/src/driver.rb
require './src/schedular'


def print_meeting(meeting)
  puts "\nMeeting Scheduled:\n\tName: #{meeting.meeting_name}\n\tStart Time: #{meeting.start_time}\n\tEnd Time: #{meeting.end_time}\n\tDuration: #{meeting.meeting_duration()}\n\tDate: #{meeting.date}\n\tAttendance Total: #{meeting.participant_list.length()} People\n\tAttendees: #{meeting.participant_list.join(', ')}\n\n"
end



first_meeting = schedule meeting {
  name 'The Cool Cids Klub'
  start 0.00
  finish 3.00
  on 10, 15, 2018
  participants [
    'me',
    'you'
  ]
}

second_meeting = schedule meeting {
  name 'Toilet Clogging Issue Meeting'
  start 8.00
  finish 8.30
  on 12, 25, 2022
  participants [
    'Janet',
    'Kyle',
    'That Dude',
    'The Janitor',
    'The Janitors Wife',
    'The Janitors Lawyer'
  ]
}

print_meeting(first_meeting)
print_meeting(second_meeting)
